README.md

```diff
@@ -65,6 +65,7 @@ The following config options are available:
 | onResult          | (path, response, target) => {} |  _.noop      | A hook to capture when the client updates a node with results from the server. Can be modified at run time by reassigning the property on a tree instance. |
 | debug             | boolean                        | false        | Debug mode will log all dispatched events and generally help debugging |
 | extend            | function                       | F.extendOn   | Used to mutate nodes internally |
+| initObject        | function                       | _.identity   | Called on the tree at initialization and on payloads before add. With `mobx`, this would be `observable` |
```

Comment:
I find it confusing when documentation refers to things that are defined/clarified later on. Comments on "Initialization" specifically are down at the end of the README

Reply:
Not sure I follow. "Initialization" isn't a `contexture-client` specific concept to be defined, it's just a word I used to mean that it's used to preprocess tree at the beginning when you call the code. Before this, we would do something like `_.flow(observable, ContextureClient(...` for mobx support and also wrap the add action's payload in observable. This is just a hook to allow that to be a deeper concept of the client (like how `snapshot` is `toJS` instead of `_.deepClone ` and `extend` is `set` instead of `F.extendOn` for mobx).

Also, it is in the `Config` section which is a subheader of `Instantiation`. The "Initialization" subheader at the end of the readme is under `Implementation Details` which explains how the internals work (which you don't need to know in order to use `contexture-client`).

Maybe the confusion is because the readme uses the words "initialization" and "instantiation" synonymously?

Reply:
No I didn't confuse those words. But when you say initialization I immediately think "at which time does this exactly happen". "Define" wasn't the best word to use in my comment.

> used to preprocess tree at the beginning when you call the code

This a bit more elaborated is exactly what I would look for upon reading the word initialization. I'm thinking this config is not really necessary for the `contextureClient` instantiation time. After all `contextureClient({...}, initObject(tree))` is equivalent to `contextureClient({..., initObject}, tree)`.

What usecase prompted you to adding `initObject` for `add`?

Reply:
That’s what we used to do. Add has always needed its payload to be an observable in the mobx case (since it needs to be an observable before it’s modified internally by initNode to set its path, defaults, etc before being added to the tree), we just used to do it on the user side by monkey patching the add action. The catalyst for making the core take this config was twofold: 1. the signature of add changed to take  another parameter (which breaks every existing implementation that overrides `add` without passing the third parameter), and 2. We added more actions that internally leverage add (that use an internal reference to the add action and so wouldn’t get the monkey patched version). It’s also possible that we end up adding more actions that need observable payloads - and at the end of the day, we’ve told the client about toJS and set, so the only thing missing is observable itself

Reply:
gonna leave this at least for now